clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -180,12 +179,12 @@ private void process(final ListOffsetsApplicationEvent event) {
      * consumer join the group if it is not part of it yet, or send the updated subscription if
      * it is already a member.
      */
-    private void process(final SubscriptionChangeApplicationEvent event) {
-        if (!requestManagers.membershipManager.isPresent()) {
-            throw new RuntimeException("Group membership manager not present when processing a " +
-                    "subscribe event");
+    private void process(final SubscriptionChangeApplicationEvent ignored) {
+        if (!requestManagers.heartbeatRequestManager.isPresent()) {
```

Comment:
I'm being a bit of a stickler here, but `RequestManagers` is designed to hold instances of `RequestManager`s, not be a holder for arbitrary objects that might be useful.

I can add the `MembershipManager` instance directly to the `ApplicationEventProcessor`.

Reply:
Well, we've certainly painted ourselves into a corner; the `RequestManagers`, `MembershipManagerImpl`, and `CommitRequestManager` are tied really closely together, so I can't refactor it without making more changes than I'd like.

Reply:
OK. Thanks for the explanation.

Reply:
Actually I think that keeping the `MembershipMgr` in the `RequestManagers` as it was before this change would be conceptually right, because we do want the `MembershipMgr` to be a regular manager, that can be polled to trigger reconciliation processes ([KAFKA-15832](https://issues.apache.org/jira/browse/KAFKA-15832)).